File: repo-root/groups/group-01-manage-field/code/managefield/src/main/java/com/jad/efreifive/managefield/vo/FieldName.java
package com.jad.efreifive.managefield.vo;

import com.jad.efreifive.managefield.exception.InvalidValueObjectException;

import java.io.Serializable;

public record FieldName(String value) implements Serializable {

    public FieldName {
        if (value == null || value.isBlank()) {
            throw new InvalidValueObjectException("field name must not be blank.");
        }
        value = value.trim();
    }

    public static FieldName of(String rawValue) {
        return new FieldName(rawValue);
    }
}

File: repo-root/groups/group-01-manage-field/code/manage-field/src/main/kotlin/fr/efrei/managefield/domain/valueobject/FieldName.kt
package fr.efrei.managefield.domain.valueobject

import java.util.Objects

/**
 * Value object representing the business name of a reservable field.
 */
@ConsistentCopyVisibility
data class FieldName private constructor(val value: String) {
    companion object {
        /**
         * Creates a field name from a raw string.
         *
         * @param raw raw field name
         * @throws IllegalArgumentException when the name is blank
         */
        fun from(raw: String): FieldName {
            val value = Objects.requireNonNull(raw, "field name must not be null").trim()
            require(value.isNotEmpty()) { "field name is required" }
            return FieldName(value)
        }
    }
}
